### PR TITLE
Adds contributing guidelines with steps

### PR DESCRIPTION
**Pull request by ankur0904**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
docs update

- **Why was this change needed?** (You can also link to an open issue here)
Fixed: #578

- **Other information**:
This Pull Request aims to enhance our contributing guidelines and includes necessary code to improve the contributor experience. The updated guidelines provide clearer instructions and expectations for potential contributors, making it easier for them to participate effectively in this project. The code changes ensure the seamless integration of these guidelines into our repository.

### CODE REVIEW

**Comment by ankur0904** [comment]
@dartpain Please review the PR.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -44,10 +44,59 @@ To run unit tests from the root of the repository, execute:
...
+8. **Collaborate:**
+   - Be responsive to comments and feedback on your PR.
+   - Make necessary updates as suggested.
+   - Once your PR is approved, it will be merged into the main repository.
+
+9. **Testing:**
+   - Before submitting a Pull Request, ensure your code passes all unit tests.
+   - To run unit tests from the root of the repository, execute:
+     ```shell
+     python -m pytest
```
Comment (dartpain) [inline CONTRIBUTING.md:93]: Please mention that only people who make backend changes need to run tests.

**Review by dartpain — CHANGES_REQUESTED** [review]
Perfect! Just one small change

**Comment by ankur0904** [comment]
@dartpain Done :)

**Review by dartpain — APPROVED** [review]

**Comment by dartpain** [comment]
Thank you!

**Comment by ankur0904** [comment]
@dartpain Thank you too.

**Pull request merged**